src/daemon/plugin.c

```diff
@@ -1306,7 +1306,7 @@ EXPORT int plugin_register_flush(const char *name, plugin_flush_cb callback,
             .data = cb,
             .free_func = plugin_flush_timeout_callback_free,
         });
-
+    sfree(cb);
```

Comment:
I think freeing it here is wrong as the data is passed to `plugin_register_complex_read` and freed after use by calling `plugin_flush_timeout_callback_free`, but I may not understand the code well enough.

Reply:
exactly, my bad. In this case it is unnecessary to call sfree() again.